code/datums/elements/venomous.dm

```diff
@@ -54,9 +58,15 @@
 		return
 	if(target.stat == DEAD)
 		return
+
 	var/final_amount_added
 	if(islist(amount_added))
 		final_amount_added = rand(amount_added[1], amount_added[2])
 	else
 		final_amount_added = amount_added
-	target.reagents?.add_reagent(poison_type, final_amount_added)
+
+	target.reagents?.add_reagent(venom.type, final_amount_added)
+	if (methods)
```

Comment:
do you need to check this? methods is set to `TOUCH` if no value is provided

Reply:
Someone might pass NONE so it kind of needs to.